### ISSUE TITLE
Cannot create feature of ID nnn because one already exists when writing to FileGDB

### ISSUE THREAD

**Issue by jontwo** [OSGeo/gdal#7724]
## Expected behavior and actual behavior.

When intersecting two layers, I get an error message like `Cannot create feature of ID 856882 because one already exists` if I output to FileGDB. Using the "SKIP_FAILURES=YES" option, the output layer contains one fewer features than at GDAL v3.5.1.

 If I output to a temporary scratch layer, there is no error and the output is the same as GDAL v3.5.1.

## Steps to reproduce the problem.

Intersect the two attached shapefile layers, either in QGIS or using a Python script such as the one below:

```
from osgeo import gdal, ogr, osr
srs = osr.SpatialReference()
srs.ImportFromEPSG(27700)
drv = gdal.GetDriverByName('OpenFileGDB')
ds = drv.Create('out.gdb', 0, 0, 0)
outlyr = ds.CreateLayer('out', srs=srs, geom_type=ogr.wkbMultiPolygon)
lyr1 = gdal.OpenEx('GB_postcode.shp')
lyr2 = gdal.OpenEx('buildings.shp')
lyr1.Intersection(lyr2, outlyr)
```

Substituting the last line for `lyr1.Intersection(lyr2, outlyr, options=["SKIP_FAILURES=YES"])` allows the intersection to run through, but one feature is missing from the output, compared to previous versions of GDAL.

## Operating system

Repro'd on both Ubuntu 22.04 and Windows 10 Pro

## GDAL version and provenance

v3.6.4 compiled from source on Ubuntu, in QGIS 3.28.6 on Windows.
[intersect_repro.zip](https://github.com/OSGeo/gdal/files/11430366/intersect_repro.zip)

**Comment by jratike80** [comment]
What do you expect to happen if you have a feature with the same unique FID (856882) in both datasets?

**Comment by rouault** [comment]
I no longer reproduce the issue with GDAL 3.7 (and likely with latest 3.6.4)
https://github.com/OSGeo/gdal/pull/6880 has fixed #6851 and now adds a input_ and method_ prefix in front of field names that appear in both datasets.
